ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transactions/TransactionMetaData.kt

```diff
@@ -0,0 +1,16 @@
+package net.corda.v5.ledger.common.transactions
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+
+/**
+ * TODO(Design question: Maybe just a simple map to make future changes easier?, Or just adding an options map to it?)
```

Comment:
Yes, I think a map + known constants would be better - then ledger specific metadata would just be some extra constants.
Could be a `Map<String, Any>` with some extension functions to fish out things? The extension functions could also be impl only.

Reply:
I have refactored it. LayeredPropertyMap seemed to be a good pattern for it.